app/javascript/admin/controllers/config_controller.js

```diff
@@ -75,28 +95,107 @@ export default class ConfigController extends Controller {
       'Cancel',
       'closeAdminConfigModal',
     );
-    if (document.querySelector('.crayons-modal')) {
-      window.scrollTo(0, 0);
-      document.body.style.height = '100vh';
-      document.body.style.overflowY = 'hidden';
-    }
+    this.positionModalOnPage();
   }
 
-  closeAdminConfigModal() {
-    this.configModalAnchorTarget.innerHTML = '';
-    document.body.style.height = 'inherit';
-    document.body.style.overflowY = 'inherit';
-  }
-
-  disableEmailAuthFromModal() {
+  disableEmailAuthFromModal(event) {
     event.preventDefault();
     emailSigninAndLoginCheckbox.checked = false;
     this.closeAdminConfigModal();
   }
 
-  disableEmailAuth() {
+  disableEmailAuth(event) {
     event.preventDefault();
     emailSigninAndLoginCheckbox.checked = false;
     this.hideEmailAuthSettings();
   }
+
+  // EMAIL AUTH FUNCTIONS END
+
+  // AUTH PROVIDERS FUNCTIONS START
+
+  enableOrEditAuthProvider(event) {
+    event.preventDefault();
+    if (event.target.dataset.buttonText === 'enable') {
+      event.target.setAttribute('data-enable-auth', 'true');
+      this.listAuthToBeEnabled();
+    }
+    const provider = event.target.dataset.authProviderEnable;
+    document
+      .querySelector(`#${provider}-auth-settings`)
+      .classList.remove('hidden');
+    event.target.classList.add('hidden');
+  }
+
+  disableAuthProvider(event) {
+    event.preventDefault();
+    const authEnableButton = document.querySelector(
+      `[data-auth-provider-enable="${event.target.dataset.authProvider}"]`,
+    );
+    authEnableButton.setAttribute('data-enable-auth', 'false');
+    this.listAuthToBeEnabled();
+    this.hideAuthProviderSettings();
+  }
+
+  authProviderModalTitle(provider) {
+    return `Disable ${provider} registration`;
+  }
+
+  authProviderModalBody(provider) {
+    return `<p>If you disable ${provider} as a registration option, people cannot create an account with ${provider}.</p><br /><p>However, people who have already created an account with ${provider} can continue to login.</p><br /><p><strong>You must update site config to save this action!</strong></p>`;
```

Comment:
This isn't true: disabling a third party auth provider **also** turns off the ability to log in to a forem with that provider.

Reply:
This is true! Hmmm ... I concur with both remedial suggestions but I don't think I can take the decision without @ben and/or @lisasy

Reply:
This has been addressed @joshpuetz 👍